core/src/main/java/apoc/util/CompressionAlgo.java

```diff
@@ -39,18 +36,18 @@
         this.decompressor = decompressor;
     }
 
-    public List<Long> compress(String string, Charset charset) throws Exception {
+    public byte[] compress(String string, Charset charset) throws Exception {
         Constructor<?> constructor = compressor.getConstructor(OutputStream.class);
         try (ByteArrayOutputStream stream = new ByteArrayOutputStream()) {
             try (OutputStream outputStream = (OutputStream) constructor.newInstance((OutputStream) stream)) {
                 outputStream.write(string.getBytes(charset));
             }
-            return convertFromBytesToList(stream.toByteArray());
+            return stream.toByteArray();
         }
     }
 
-    public String decompress(List<Long> byteArray, Charset charset) throws Exception {
-        try (ByteArrayInputStream stream = new ByteArrayInputStream(convertFromListToBytes(byteArray))) {
+    public String decompress(byte[] byteArray, Charset charset) throws Exception {
+        try (ByteArrayInputStream stream = new ByteArrayInputStream(byteArray)) {
             Constructor<?> constructor = decompressor.getConstructor(InputStream.class);
```

Comment:
nit: you could move that to the top, to be consistent with `compress`
that way, you'd need only 1 try-with-resources block

Reply:
Changed 👍